src/generic/eigen_solver.h

```diff
@@ -162,15 +224,33 @@ namespace oomph
     }
 
     /// Solve the eigen problem
+    void solve_eigenproblem_legacy(Problem* const& problem_pt,
+                                   const int& n_eval,
+                                   Vector<std::complex<double>>& eigenvalue,
+                                   Vector<DoubleVector>& eigenvector);
+
+
+    /// Solve the real eigenproblem that is assembled by elements in
+    /// a mesh in a Problem object. Note that the assembled matrices include the
+    /// shift and are real. The eigenvalues and eigenvectors are,
+    /// in general, complex. Eigenvalues may be infinite and are therefore
+    /// returned as
+    /// \f$ \lambda_i = \alpha_i / \beta_i \f$ where \f$ \alpha_i \f$ is complex
+    /// while \f$ \beta_i \f$ is real. The actual eigenvalues may then be
+    /// computed by doing the division, checking for zero betas to avoid NaNs.
+    /// There's a convenience wrapper to this function that simply computes
+    /// these eigenvalues regardless. That version may die in NaN checking is
+    /// enabled (via the fenv.h header and the associated feenable function).
+    /// At least n_eval eigenvalues are computed.
     void solve_eigenproblem(Problem* const& problem_pt,
                             const int& n_eval,
-                            Vector<std::complex<double>>& eigenvalue,
-                            Vector<DoubleVector>& eigenvector);
-
-    /// Use the eigensolver to find the eigenvalues of a given matrix
-    // void find_eigenvalues(const DoubleMatrixBase &A, const int &n_eval,
-    //                      Vector<std::complex<double> > &eigenvalue,
-    //                      Vector<Vector<double> > &eigenvector);
+                            Vector<std::complex<double>>& alpha,
+                            Vector<double>& beta,
+                            Vector<Vector<std::complex<double>>>& eigenvector)
+    {
+      oomph_info << "Broken, but then don't we want arpack to go anyway?\n";
```

Comment:
Yes, we should eventually remove ARPACK. Maybe indicate that it is deprecated. Do we have a standard mechanism for that? I can't remember.

Reply:
If you're happy for it to go then we should just make it go. Shall I do the honours?

Reply:
I've made (in the updated pull request) lapack's QZ the default eigensolver but kept the code for arpack around so you can still use it to assess what features of it you need. There are some that are only available in arpack and you may want/need to enable them in the anasazi wrapper. 

Done.

Reply:
Made lapack QZ the default eigensolver. Kept arpack around so you can revisit it and possibly port any capabilities across to anazsazi. All self-tests pass without it and there are no longer any demo drivers that rely on arpack.

Done.